project_cost_report/report.py

```diff
@@ -437,3 +439,100 @@ def get_foldable_categories(self, project_id):
         res = super().get_foldable_categories(project_id)
         res['outsourcing'] = [False]
         return res
+
+
+class ProjectCostReportWithTotalCost(models.TransientModel):
+
+    _inherit = 'project.cost.report'
+
+    def _get_rendering_variables(self, project, report_context):
+        """Add the total of costs to the rendering variables."""
+        res = super()._get_rendering_variables(project, report_context)
+        res.update({
+            'total_cost': float_round(
+                res['product_total'] + res['time_total'] + res['outsourcing_total'], 2
+            )
+        })
+        return res
+
+
+def purchase_line_is_pending_invoice(line: models.Model) -> bool:
+    """Determine whether a purchase line is pending an invoice.
+
+    :param line: a purchase.order.line singleton
+    """
+    precision = line.env['decimal.precision'].precision_get('Product Unit of Measure')
+    less_units_invoiced_than_purchased = (
+        float_compare(line.qty_invoiced, line.product_qty, precision_digits=precision) == -1
+    )
+    return less_units_invoiced_than_purchased
+
+
+def get_purchase_line_pending_qty(line: models.Model) -> float:
+    """Get the quantity of units pending invoices from a purchase line.
+
+    :param line: a purchase.order.line singleton
+    """
+    precision = line.env['decimal.precision'].precision_get('Product Unit of Measure')
+    return float_round(line.product_qty - line.qty_invoiced, precision_digits=precision)
+
+
+class PendingPurchaseOrder:
+    """Represents a purchase order in the cost report.
+
+    This object is used to display the share of amount pending invoice
+    that is bound to the current project.
+
+    :param order: the purchase order represented.
+    :param project: the project for which to render the report.
+    """
+
+    def __init__(self, order: models.Model, project: models.Model):
+        self.order = order
+        self.lines_pending_invoices = order.order_line.filtered(
+            lambda l: purchase_line_is_pending_invoice(l) and
+            l.account_analytic_id == project.analytic_account_id
+        )
+        self.total = float_round(
+            sum(l.price_unit * get_purchase_line_pending_qty(l)
+                for l in self.lines_pending_invoices),
+            2
+        )
+
+
+class ProjectCostReportWithPendingInvoices(models.TransientModel):
+
+    _inherit = 'project.cost.report'
+
+    def _get_pending_purchase_order_lines(self, project):
+        """Get the purchase order lines with unreceived invoices.
+
+        :param project: the project.project record
+        :rtype: purchase.order.line
+        """
+        domain = [
+            ('account_analytic_id', '=', project.analytic_account_id.id),
+            ('order_id.state', 'in', ('purchase', 'done')),
+        ]
+        lines = self.env['purchase.order.line'].search(domain)
+        return lines.filtered(lambda l: purchase_line_is_pending_invoice(l))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Lambda may not be necessary](https://app.codacy.com/app/numigi/odoo-project-addons/pullRequest?prid=3730559)